File: RutujaMadane/Microsoft/Day1/bulls_cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        int n=secret.size();
        vector<int>s(10,0);
        vector<int>g(10,0);
        int B=0;
        int C=0;
        for(int i=0;i<n;i++){
            if(secret[i]==guess[i]){
                B++;
            }
            else{
                s[secret[i]-'0']++;
                g[guess[i]-'0']++;
            }
        }
        for(int i=0;i<10;i++){
            C+=min(s[i],g[i]);
        }
        string result="";
        result+=to_string(B);
        result+='A';
          result+=to_string(C);
        result+='B';

        return result;

    }
};

File: Harishri/Microsoft/Bulls_Cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        lst1 = list(secret)
        lst2 = list(guess)
        x = 0
        for i in range(len(lst2)):
            if lst1[i] == lst2[i]:
                x += 1
        y = len(lst1) - sum((Counter(lst1) - Counter(lst2)).values()) - x
        res = ''.join(str(x)+'A'+str(y)+'B')
        return res
